### ISSUE TITLE
Auto append file extension when user renames file for download

### ISSUE THREAD

**Issue by johnbotris** [tutao/tutanota#2675]
When saving attachments, you can rename the file in the file chooser dialog. If you overwrite the extension, then the file will be saved without a file extension. One would generally expect the extension to be preserved automatically in this case. This is also a bigger problem in Windows as Windows identifies a file's type through it's extension alone, and won't inspect it's contents to try and determine what kind of file it is, breaking double-click to open, etc.

In short, when a user renames a file we should preserve the original extension.

**Comment by johnbotris** [comment]
after some investigation, it might be that we don't necessarily want to automatically append the lost file extension. Firefox doesn't do this, for example, although Edge does. 

We would also run the risk of inadvertently overwriting existing files, since electron would warn the user based on the name they provided, but afterwards we would be changing the name behind the scenes, which could then be a duplicate.
